test/integration/scheduler/queue_test.go

```diff
@@ -170,6 +172,108 @@ func TestServiceAffinityEnqueue(t *testing.T) {
 	}
 }
 
+func TestCoreResourceEnqueue(t *testing.T) {
+	cfg := configtesting.V1beta2ToInternalWithDefaults(t, v1beta2.KubeSchedulerConfiguration{
+		Profiles: []v1beta2.KubeSchedulerProfile{{
+			SchedulerName: pointer.StringPtr(v1.DefaultSchedulerName),
+			Plugins: &v1beta2.Plugins{
+				PreFilter: v1beta2.PluginSet{
+					Enabled: []v1beta2.Plugin{
+						{Name: noderesources.FitName},
+					},
+				},
+				Filter: v1beta2.PluginSet{
+					Enabled: []v1beta2.Plugin{
+						{Name: tainttoleration.Name},
+						{Name: noderesources.FitName},
+					},
+				},
+			},
+		}},
+	})
+
+	// Use zero backoff seconds to bypass backoffQ.
+	testCtx := testutils.InitTestSchedulerWithOptions(
+		t,
+		testutils.InitTestAPIServer(t, "core-res-enqueue", nil),
+		nil,
+		scheduler.WithProfiles(cfg.Profiles...),
+		scheduler.WithPodInitialBackoffSeconds(0),
+		scheduler.WithPodMaxBackoffSeconds(0),
+	)
+	testutils.SyncInformerFactory(testCtx)
+	// It's intended to not start the scheduler's queue, and hence to
+	// not start any flushing logic. We will pop and schedule the Pods manually later.
+	defer testutils.CleanupTest(t, testCtx)
+
+	cs, ns, ctx := testCtx.ClientSet, testCtx.NS.Name, testCtx.Ctx
+	// Create one Node with a taint.
+	node := st.MakeNode().Name("fake-node").Capacity(map[v1.ResourceName]string{v1.ResourceCPU: "2"}).Obj()
+	node.Spec.Taints = []v1.Taint{{Key: "foo", Effect: v1.TaintEffectNoSchedule}}
+	if _, err := cs.CoreV1().Nodes().Create(ctx, node, metav1.CreateOptions{}); err != nil {
+		t.Fatalf("Failed to create Node %q: %v", node.Name, err)
+	}
+
+	// Create two Pods that are both unschedulable.
+	// - Pod1 is a best-efforts Pod, but doesn't have the required toleration.
+	// - Pod2 has the required toleration, but requests a large amount of CPU resource that the node cannot fit.
```

Comment:
I think Pod2 has no tolerations, but not affect the finally test results.

Reply:
Yes, I should have updated the comment. Done right now.

My original thought was to add a toleration for pod2, but it turned out with no extra code coverage. So I intentionally leave it without toleration, but with excessive pod req - that will cover the usage of `preCheckForNode()`.